src/txn/txn_recover.c

```diff
@@ -556,7 +558,8 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 	 */
 	WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
-done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
+done:	FLD_SET(conn->log_flags,
+	    WT_CONN_LOG_RECOVER_DIRTY | WT_CONN_LOG_RECOVER_DONE);
```

Comment:
We only set it above if we detect `needs_rec` (i.e. we detect an unclean shutdown), but recovery is single-threaded and I simply unconditionally clear it.  It seemed messier in the code to conditionalize that clear.